client/components/data-label/label.directive.js

```diff
@@ -0,0 +1,94 @@
+function linkFn(type) {
```

Comment:
I think these functions can be put at the bottom of the file without change in functionality. I would prefer to see the `angular.module` call at the top if it works.